backend/core/auth.py

```
@@ -0,0 +1,138 @@
+            "teacher_id": profile_data.get('teacher_id'),
+            "grade_level": profile_data.get('grade_level'),
+            "department": profile_data.get('department')
+        }
+    
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Error fetching user profile: {str(e)}"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)